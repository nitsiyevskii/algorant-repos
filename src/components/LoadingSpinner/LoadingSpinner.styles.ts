import { createStyles } from "src/utils/createStyles";

export const useStyles = createStyles((theme) => ({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: theme.colors.background.primary,
  },
}));

