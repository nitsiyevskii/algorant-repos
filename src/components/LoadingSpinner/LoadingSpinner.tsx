import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import { useTheme } from 'src/hooks/useTheme';
import { useStyles } from './LoadingSpinner.styles';

interface LoadingSpinnerProps {
  size?: 'small' | 'large';
  color?: string;
}

export const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({
  size = 'large',
  color,
}) => {
  const theme = useTheme();
  const styles = useStyles();

  return (
    <View style={styles.container} testID="loading-spinner">
      <ActivityIndicator
        size={size}
        color={color || theme.colors.core.primary}
        testID="activity-indicator"
      />
    </View>
  );
};

