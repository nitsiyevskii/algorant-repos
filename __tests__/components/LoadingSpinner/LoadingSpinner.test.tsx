import React from 'react';
import { render } from '@testing-library/react-native';
import { LoadingSpinner } from 'src/components/LoadingSpinner/LoadingSpinner';

describe('LoadingSpinner', () => {
  it('renders correctly with default props', () => {
    const { toJSON } = render(<LoadingSpinner />);
    expect(toJSON()).toBeTruthy();
  });

  it('renders with small size', () => {
    const { getByTestId } = render(<LoadingSpinner size="small" />);
    const spinner = getByTestId('activity-indicator');
    expect(spinner.props.size).toBe('small');
  });

  it('renders with large size', () => {
    const { getByTestId } = render(<LoadingSpinner size="large" />);
    const spinner = getByTestId('activity-indicator');
    expect(spinner.props.size).toBe('large');
  });

  it('renders with custom color', () => {
    const { getByTestId } = render(<LoadingSpinner color="#FF0000" />);
    const spinner = getByTestId('activity-indicator');
    expect(spinner.props.color).toBe('#FF0000');
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<LoadingSpinner />);
    expect(toJSON()).toMatchSnapshot();
  });
});

