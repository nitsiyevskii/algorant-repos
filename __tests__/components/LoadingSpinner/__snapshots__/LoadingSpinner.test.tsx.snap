// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F2F2F7",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="loading-spinner"
>
  <ActivityIndicator
    color="#007AFF"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
